remove other RBin unused functions and make others static

ret2libc opened the pull request:
It also introduces a WARNING when plugins do not implement the load_buffer method

radare commented on libr/bin/obj.c:
```diff
@@ -8,17 +8,135 @@
+		r_list_free (classes);
+		return NULL;
+	}
+	return classes;
```
Maybe we need to fill o->classes directly here instead of returning a list, because this function is appending them and returning which is conceptually wrong and may lead to dupped data

ret2libc replied:
actually... it seems like classes will be always NULL :S I don't see any r_list_ on the `classes` variable in `classes_from_symbols`. Do you see where this is filled? It seems all this thing with classes is useless :/ I'd just keep the r_list_foreach in the body and remove the rest...

radare replied:
ok

radare commented on libr/bin/bfile.c:
```diff
@@ -885,6 +744,61 @@ static int is_data_section(RBinFile *a, RBinSection *s) {
+	RBinString *ptr;
+	RListIter *it;
+
+	if (!bf || !bf->buf) {
```
R return if fail this

radare commented on libr/bin/bfile.c:
```diff
@@ -885,6 +744,61 @@ static int is_data_section(RBinFile *a, RBinSection *s) {
 }
 
+static void get_strings_range(RBinFile *bf, RList *list, int min, int raw, ut64 from, ut64 to) {
+	RBinPlugin *plugin = r_bin_file_cur_plugin (bf);
```
Do this after checking for null

radare commented on libr/bin/bin.c:
```diff
@@ -1285,6 +1285,20 @@ R_IPI void r_bin_class_free(RBinClass *c) {
+
+	RBinClass *c;
+	RListIter *iter;
+	RList *list = binfile->o->classes;
```
Unnecessary variable

radare commented on libr/bin/bin.c:
```diff
@@ -1285,6 +1285,20 @@ R_IPI void r_bin_class_free(RBinClass *c) {
+	RBinClass *c;
+	RListIter *iter;
+	RList *list = binfile->o->classes;
+	r_list_foreach (list, iter, c) {
```
Add a warning message because this is o(n) and we shohld use a kv to resolve this faster

radare commented on libr/bin/bfile.c:
```diff
@@ -953,67 +867,12 @@ R_IPI RList *r_bin_file_get_strings(RBinFile *a, int min, int dump, int raw) {
 		}
 	} else {
 		if (a) {
-			r_bin_file_get_strings_range (a, ret, min, raw, 0, a->size);
```
Ideally this functionality should be implemented in rsearch. But this change doesnt fits in this pr. Just keep it in mind for future changes

ret2libc replied:
ok good to know

radare commented on libr/bin/obj.c:
```diff
@@ -8,17 +8,135 @@
+	if (mem && mem->mirrors) {
+		mem->mirrors->free = mem_free;
+		r_list_free (mem->mirrors);
+		mem->mirrors = NULL;
```
No need to null

ret2libc replied:
Are you sure? I'm just keeping the code as it was, I didn't really looked at it, just moved the function from one place to another.

radare commented on libr/bin/obj.c:
```diff
@@ -8,17 +8,135 @@
+	free (mem);
+}
+
+static void object_delete_items(RBinObject *o) {
```
This is called only once from objfree. What about moving all this code in there and remove the function?

ret2libc replied:
I'm actually ok having a separate function. I tink it makes things simpler to read.

radare commented on libr/bin/bfile.c:
```diff
@@ -885,6 +744,61 @@ static int is_data_section(RBinFile *a, RBinSection *s) {
+			if (bf->rbin->verbose) {
+				eprintf ("WARNING: bin_strings buffer is too big "
+					"(0x%08" PFMT64x
+					")."
```
Ugly indent

ret2libc commented:
Ok, I think this is enough for 1 PR. Let's see what travis says and if you @radare have other comments.

Some things to note:
- i've added a r_assert to be sure that "any" plugin is always available
- implemented load_buffer in the "any" plugin, by creating an empty r_buf that is never used

radare commented on libr/bin/obj.c:
```diff
@@ -8,21 +8,148 @@
+	return NULL;
+}
+
+static RList *classes_from_symbols(RBinFile *bf, RBinObject *o) {
```
Just pass RBinFile, and dereference the RBinObject from the bf.

at some point we shuold merge those two structs to make it less confusing

radare commented on libr/bin/obj.c:
```diff
@@ -8,21 +8,148 @@
+	RBinSymbol *sym;
+	RListIter *iter;
+	RList *symbols = o->symbols;
+	RList *classes = o->classes;
```
what about not taking this and always create a new list?, the one who calls this function should be in charge to fill o->classes?

For consistency maybe it should be like this,otherwise the function name may change. If we decide to just updae the binobj inside each of those calls instead of returning a list. im fine too. but it can be useful at some point to have access to those apis, not just having them internally.

radare commented on libr/bin/obj.c:
```diff
@@ -8,21 +8,148 @@
 }
 
+static char *swiftField(const char *dn, const char *cn) {
+	char *p = strstr (dn, ".getter_");
```
dn and cn can be null

radare commented on libr/bin/obj.c:
```diff
@@ -8,21 +8,148 @@
+	return true;
+}
+
 R_IPI RBinObject *r_bin_object_new(RBinFile *binfile, RBinPlugin *plugin, ut64 baseaddr, ut64 loadaddr, ut64 offset, ut64 sz) {
```
what do you think about renaming r_bin_object_ into r_bin_obj_ or r_binobj_ ?

radare commented on libr/bin/bfile.c:
```diff
@@ -411,110 +341,48 @@ R_API bool r_bin_file_object_new_from_xtr_data(RBin *bin, RBinFile *bf, ut64 bas
-	if (sz == UT64_MAX) {
-		return NULL;
+static RBinPlugin *get_plugin(RBin *bin, const char *pluginname, const ut8 *bytes, ut64 sz) {
+	RBinPlugin *plugin;
```
defiene and assign in the same line

radare commented on libr/bin/obj.c:
```diff
@@ -8,21 +8,148 @@
+		o->mem->free = mem_free;
+	}
+	r_list_free (o->mem);
+	o->mem = NULL;
```
remove all those = NULL

radare commented on libr/bin/obj.c:
```diff
@@ -8,21 +8,148 @@
+	o->lines = NULL;
+	o->info = NULL;
+	o->kv = NULL;
+	for (i = 0; i < R_BIN_SYM_LAST; i++) {
```
and this loop too

radare replied:
i mean the = NULL inside the loop

ret2libc commented:
@radare I think i've addressed all comments. I *think* i've also fixed the `classes_from_symbols` function.

radare commented:
Rbinfile holds one rbinobj inside. So they are basically the same. I mean you cannot think in them as separate identities. We should always use one of them and imho it shoukd ve rbinfile, even if we wanr to rename it later

> On 7 Nov 2018, at 09:26, Riccardo Schirone <notifications@github.com> wrote:
> 
> @ret2libc commented on this pull request.
> 
> In libr/bin/obj.c:
> 
> > +		}
> +	}
> +	if (r_list_empty (classes)) {
> +		r_list_free (classes);
> +		return NULL;
> +	}
> +	return classes;
> +}
> +
> +static bool file_object_add(RBinFile *binfile, RBinObject *o) {
> +	r_return_val_if_fail (binfile && o, false);
> +	r_list_append (binfile->objs, o);
> +	r_bin_file_set_cur_binfile_obj (binfile->rbin, binfile, o);
> +	return true;
> +}
> +
>  R_IPI RBinObject *r_bin_object_new(RBinFile *binfile, RBinPlugin *plugin, ut64 baseaddr, ut64 loadaddr, ut64 offset, ut64 sz) {
> I'm ok with this change, but I'd wait a bit before doing it, since we are probably going to merge/remove/change either RBinFile or RBinObject.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

ret2libc commented:
Perfect, that's one more reason to do the renaming later rather than now.
Are you ok merging this?
